src/store/modules/nav/reducer.ts

```diff
@@ -1,13 +1,61 @@
 import { Reducer } from 'redux';
+import * as R from 'ramda';
 
 import { NavState, Action } from './types';
 import initialState from './state';
-import Navigator from '../../../view/navigation/navigator';
+import Navigator, { Route } from '../../../view/navigation/navigator';
+import { NavigationActions, NavigationState } from 'react-navigation';
 
 const navReducer: Reducer<NavState> = (state = initialState, action: Action) => {
-  const nextState = Navigator.router.getStateForAction(action, state);
+  switch (action.type) {
 
-  return nextState || state;
+    case 'LOGOUT': {
+      let routes = state.routes.filter(r => r.routeName === Route.LOGIN);
+      let index = 0;
+
+      return { ...state, routes, index };
+    }
+
+    case 'LOGIN': {
+      let navigateAction = NavigationActions.navigate({ routeName: Route.HOME });
+      let nextState = Navigator.router.getStateForAction(navigateAction, state) as NavigationState;
+      let routes = R.remove(0, 1, nextState.routes);
+      let index = nextState.index - 1;
+
+      return { ...nextState, routes, index };
+    }
+
+    case 'NAV/GO_TO_WITHOUT_HISTORY': {
+      let subtractLength = 2;
+      let navigateAction = NavigationActions.navigate({ routeName: action.route });
+      let nextState = Navigator.router.getStateForAction(navigateAction, state) as NavigationState;
+      let routes = R.remove(nextState.routes.length - subtractLength, 1, nextState.routes);
+      let index = nextState.index - 1;
+      if (routes[index].routeName === routes[index - 1].routeName) {
```

Comment:
I don't think so, the worst that could happen is someone calls it on home or login and it wipes the history for that page, which would be silly but it could happen.